lua/nvim-treesitter/parsers.lua

```diff
@@ -348,6 +348,16 @@ list.haskell = {
   }
 }
 
+list.hcl = {
+  install_info = {
+    url = "https://github.com/MichaHoffmann/tree-sitter-hcl",
+    files = {"src/parser.c", "src/scanner.cc"},
+    branch = "main"
+  },
+  filetype = "hcl",
+  used_by = {"tf", "tfvars"}
```

Comment:
Should that not contain `terraform`? [hashivim/vim-terraform](https://github.com/hashivim/vim-terraform) sets `ft=terraform` for `*.tf` files.

Reply:
Is the filetype canonical? Maybe it would be better to write a ftdetect function that adds the `hcl` filetype for terraform files. What do you think?

Reply:
I am really not sure, but I remember that tf filetyoe is something different to default vim than terraform.

Reply:
> hashivim/vim-terraform sets ft=terraform for *.tf files.

I do most of the maintenance on hashivim/vim-terraform.  Happy to consider tweaking the filetype we use, if that's the Right Thing To Do.  Especially happy to consider proposals made in the form of MRs...

I don't think that vim-terraform should use `hcl` - the plugin provides some genuinely terraform-specific features.  But perhaps it would make sense to use `hcl.terraform`?  

(I'm not entirely sure what all the implications of that are, but it sounds plausible and I'm open to guidance!)

>  remember that tf filetyoe is something different to default vim than terraform.

Yes, TinyFugue

Reply:
`hcl.terraform` sounds right to me. I created https://github.com/hashivim/vim-terraform/pull/168 and will drop the `used_by` fields from the configuration here.

Reply:
Since other plugins set the filetype to `terraform` or `nomad` ( e.g. https://github.com/jvirtanen/vim-hcl/blob/master/ftdetect/hcl.vim ), I added `used_by: { "terraform", "nomad", "packer" }` to the parser configuration and closed the `vim-terraform` PR again. This seemed to be the most consensual configuration.